datalad/support/gitrepo.py

```diff
@@ -3109,12 +3154,14 @@ def _diffstatus_get_state_props(self, f, from_state, to_state,
         Parameters
         ----------
         f : Path
-        from_state : dict
-        to_state : dict
+        from_state : dict or None
+        to_state : dict or None
```

Comment:
what does it mean for state to be `None` -- current worktree? might be worth a docstring

Reply:
`None` means that there is not state for this path in either the `to` or `from` part, because that path is added or removed, for example.